src/Vehicle/Vehicle.cc

```diff
@@ -263,6 +266,9 @@ void Vehicle::_sendMessage(mavlink_message_t message)
             static uint8_t messageKeys[256] = MAVLINK_MESSAGE_CRCS;
             mavlink_finalize_message_chan(&message, mavlink->getSystemId(), mavlink->getComponentId(), link->getMavlinkChannel(), message.len, messageKeys[message.msgid]);
             
+            // Give the plugin a chance to adjust
+            _firmwarePlugin->adjustMavlinkMessage(&message);
+            
```

Comment:
In libmavconn i done that way:
https://github.com/mavlink/mavros/blob/master/libmavconn/src/interface.cpp#L79
https://github.com/mavlink/mavros/blob/master/libmavconn/include/mavconn/msgbuffer.h#L43

Normally finalize done by pack/encode step.